### DISCUSSION TITLE
Large payload, read and processed piece by piece

### DISCUSSION THREAD

**Discussion by tiolan** [uNetworking/uWebSockets#1365] [Q&A]
Hey, I have seen similar discussions here, though my usecase seems to be a bit different.

I wonder if uwebsockets supports the following usecase.
A client has a large blob of binary data that it wants to send to the server via a POST.
The server runs on a constrained device and should be able to read and process the headers, but should not try to read (from network stack) e.g. decode the payload, nor should it read the entire payload into it's RAM.
The binary payload is larger than the read buffer in the server.
After processing the headers, I need to be able to read the payload "byte by byte from the wire" and then process these chunks (I do not mean actual http chunking here).

So what I would need technically in the server is something like a read_payload(void*, size_t) callback/function, that would be called multiple times, in order to "pump" the payload chunk by chunk.

Is that supported?

**Comment by ddevienne** [comment]
Sounds like you want access to the [frames](https://datatracker.ietf.org/doc/html/rfc6455#section-5) that messages are made of, then, no?
I don't recall uWebSocket giving access to the frames, but I could be wrong.

**Reply by tiolan** [reply]
I would use the library as an HTTP only server, not using websockets at all (I understood that's possible).
So I guess no, I do not want to access frames.
I just want to control how many bytes the library reads from the wire.
Something like a Ring-Buffer for the payload, that I can control to be filled and emptied "on the fly".
